.github/dependabot.yml

```
@@ -0,0 +1,19 @@
+version: 2
+updates:
+- package-ecosystem: bundler
+  directory: "/"
+  schedule:
+    interval: daily
+    time: "11:00"
+    timezone: Europe/London
+  open-pull-requests-limit: 3
+  versioning-strategy: increase-if-necessary
```

```suggestion
  versioning-strategy: increase
```

Once we've updated and started building on a new version, we don't want it to be possible to accidentally run on older versions.

---

I don't think either strategy will allow us to go backwards would it? I interpret `increase-if-necessary` as making updates only when needed and `increase` updating regardless of whether they are needed?

This script only commits the current logic as a config file. It seems to me that `increase-if-necessary` is good enough to start and it will lower the amount of updates to those that are necessary. Not including security updates of course.

---

My understanding is that if you specify a version range in your `Gemfile` `increase` will always increase that version range, while `increase-if-necessary` will only do so if the new version isn't in the range. That means deleting a lockfile and installing fresh or (maybe) running `bundle update` might result in older versions returning. Feels safer to just always `increase`?

---

Hopefully deleting lockfiles isn't something we would need to do as they are meant to provide us that level of stability we are worried about?

Finding it hard to determine the interpretation of these options with confidence. 

## increase-if-necessary

Dependabot it generated https://github.com/dxw/rails-template/pull/140.

## increase

Dependabot generated this https://github.com/dxw/rails-template/pull/127/files which changes the Gemfile range too. This is interesting because I thought `~> 4.1` meant `< 5.0`, which would undermine the use of range?

I've had a look at other projects we have configured and actually most of them actually use `auto`. As this repository is a template it might make sense to have `increase` since we don't have the same compatibility concerns of a real app. 

On the basis that this is the most popular configuration we use and that it _only_ seeks to change the lockfiles I am now in favour or `auto` as a default.

---

Since we have 3 approvals now I am going to back `auto` and be open to changing it again later if we need.